app-backend/db/src/test/scala/com/azavea/rf/database/PlatformDaoSpec.scala

```diff
@@ -276,4 +276,70 @@ class PlatformDaoSpec extends FunSuite with Matchers with Checkers with DBTestCo
       }
     }
   }
+
+  test("list teams a platform user belongs to or can see due to organization memberships") {
+    check {
+      forAll{
+        (
+          orgCreate1: Organization.Create,
+          orgCreate2: Organization.Create,
+          orgCreate3: Organization.Create,
+          teamCreate1: Team.Create,
+          teamCreate2: Team.Create,
+          teamCreate3: Team.Create,
+          userCreate: User.Create,
+          teamNamePartial: SearchQueryParameters
+        ) => {
+          val creatTeamsIO =  for {
+            // Team# is in Org#
+            // User belongs to Org1 and Team1
+            orgUserInserted1 <- insertUserAndOrg(userCreate, orgCreate1, true)
+            (org1, user) = orgUserInserted1
+            teamInsert1 <- TeamDao.create(fixupTeam(
+              fixTeamName(teamCreate1, teamNamePartial), org1, user))
+            _ <- UserGroupRoleDao.create(
+              UserGroupRole.Create(
+                user.id,
+                GroupType.Team,
+                teamInsert1.id,
+                GroupRole.Member
+              ).toUserGroupRole(user, MembershipStatus.Approved))
+
+            // User belongs to Org2 but not Team2
+            org2 <- OrganizationDao.createOrganization(orgCreate2)
+            _ <- UserGroupRoleDao.create(
+              UserGroupRole.Create(
+                user.id,
+                GroupType.Organization,
+                org2.id,
+                GroupRole.Member
+              ).toUserGroupRole(user, MembershipStatus.Approved))
+            teamInsert2 <- TeamDao.create(fixupTeam(
+              fixTeamName(teamCreate2, teamNamePartial), org2, user))
+
+            // User belongs to Team3 but not Org3
+            org3 <- OrganizationDao.createOrganization(orgCreate3)
+            teamInsert3 <- TeamDao.create(fixupTeam(
+              fixTeamName(teamCreate3, teamNamePartial), org3, user))
+            _ <- UserGroupRoleDao.create(
+              UserGroupRole.Create(
+                user.id,
+                GroupType.Team,
+                teamInsert3.id,
+                GroupRole.Member
+              ).toUserGroupRole(user, MembershipStatus.Approved))
+
+          } yield (teamInsert1, teamInsert2, teamInsert3, user)
+
+          val filterTeamIO = creatTeamsIO flatMap {
+            case (teamInsert1: Team, teamInsert2: Team, teamInsert3: Team, user: User) =>
+              PlatformDao.listPlatformUserTeams(user, teamNamePartial)
+          }
+          val teams = filterTeamIO.transact(xa).unsafeRunSync
+
+          teams.length == 3
```

Comment:
Right now it's hard for this test to fail even if search isn't working.
It happens to be the case based on the way the test is written that `length == 3` is equivalent to
"all of the returned teams include the search string", but I think a better test of this behavior would be

- insert some teams unrelated to the search string
- confirm all of the returned teams have names that include the search string

Reply:
Based on how `fixTeamName` is written, I think that's just inserting teams without doing `fixTeamName` right?